jiant/preprocess.py

```diff
@@ -396,7 +396,19 @@ def _get_task(name, args, data_path, scratch_path):
     """ Build or load a single task. """
     assert name in TASKS_REGISTRY, f"Task '{name:s}' not found!"
     task_cls, rel_path, task_kw = TASKS_REGISTRY[name]
+
+    task_pkls = glob.glob(os.path.join(scratch_path, "tasks", "%s*" % name))
     pkl_path = os.path.join(scratch_path, "tasks", f"{name:s}.{args.tokenizer:s}.pkl")
+
+    # There should only exist one task pickle per experiment directory with only one
+    # preprocessing, thus check that this is the case if the task pickle directory exists.
+    assert (
+        not os.path.isdir(os.path.join(scratch_path, "tasks"))
+        or len(task_pkls) == 0
+        or args.tokenizer in task_pkls[0]
```

Comment:
It seems plausible that for future models, one tokenizer name might be a substring of another, (maybe 'bert-base-cased' and 'multilingual-bert-base-cased'), in which case this check will pass. Mind extracting the exact substring? (Can you do this with pkl_path?)

Reply:
+1. This kind of bug happens more than you think :)